pennylane/templates/subroutines/qdrift.py

```diff
@@ -0,0 +1,227 @@
+    of the Hamiltonian. With probability :math:`p_j` we will add to the product the operator
+    :math:`\exp{(\frac{i \lambda H_j}{N})}`, where :math:`\lambda = \sum_{j=1}^{N} |h_j|`.
+    On the other hand, :math:`p_j` is calculated as :math:`p_j = \frac{|h_j|}{\lambda}`.
+    The number of terms to be added to the product will be entered in the class constructor.
+
+    Args:
+        hamiltonian (Union[~.Hamiltonian, ~.Sum]): The Hamiltonian of the system.
+        time (complex): The time for which the system evolves.
+
+    Keyword Args:
+        n (int): The number of terms to be added to the product formula. Default is 1.
+        seed (int): The seed for the random number generator. Default is 42.
+    Raises:
+        TypeError: The 'hamiltonian' is not of type ~.Hamiltonian, or ~.Sum.
+        ValueError: One or more of the terms in 'hamiltonian' are not Hermitian.
+        ValueError: The number of samples 'n' is not an integer.
+
+    **Example**
+
+    .. code-block:: python3
+
+        coeffs = [0.25, 0.75]
+        ops = [qml.PauliX(0), qml.PauliZ(0)]
+        H = qml.dot(coeffs, ops)
+
+        dev = qml.device("default.qubit", wires=2)
+        @qml.qnode(dev)
+        def my_circ():
+            # Prepare some state
+            qml.Hadamard(0)
+
+            # Evolve according to H
+            qml.QDrift(H, time=1.2, n = 10)
+
+            # Measure some quantity
+            return qml.probs()
+
+    >>> my_circ()
+    [0.52825127 0.         0.47174873 0.        ]
+
+
+    .. details::
+        :title: Usage Details
+
+        We can also compute the gradient with respect to the
+        evolution time:
+
+        .. code-block:: python3
+
+            @qml.qnode(dev)
+            def my_circ(time):
+                # Prepare H:
+                H = qml.dot([0.2, -0.1], [qml.PauliY(0), qml.PauliZ(1)])
+
+                # Prepare some state
+                qml.Hadamard(0)
+
+                # Evolve according to H
+                qml.QDrift(H, time, n=10, seed=10)
+
+                # Measure some quantity
+                return qml.expval(qml.PauliZ(0) @ qml.PauliZ(1))
+
+
+        >>> args = qnp.array([1.23])
+        >>> print(qml.grad(my_circ)(*tuple(args)))
+        0.27980654844422853
+    """
+
+    def __init__(  # pylin: disable=too-many-arguments
+        self, hamiltonian, time, n=1, seed=None, id=None
+    ):
+        r"""Initialize the QDrift class"""
+
+        if isinstance(hamiltonian, qml.Hamiltonian):
+            coeffs, ops = hamiltonian.terms()
+            hamiltonian = qml.dot(coeffs, ops)
+
+        if not isinstance(hamiltonian, Sum):
+            raise TypeError(
+                f"The given operator must be a PennyLane ~.Hamiltonian or ~.Sum got {hamiltonian}"
+            )
+
+        self._hyperparameters = {
+            "n": n,
+            "seed": seed,
+            "base": hamiltonian,
+        }
+        super().__init__(time, wires=hamiltonian.wires, id=id)
+
+    def _flatten(self):
```

Comment:
If qdrift is not differentiable with respect to the coefficients of the Hamiltonian, I believe that the flatten and unflatten functions can be deleted